From 5d77935a3008948e6c222a414496fa1484653fb1 Mon Sep 17 00:00:00 2001
From: Min He <min.he@intel.com>
Date: Fri, 7 Sep 2018 15:15:38 +0800
Subject: [PATCH 1632/1667] drm/i915: to limit the supported modifiers for
 plane restriction

In GVT-g environment, to ensure all the OS's have enough DDB to
display, GVT-g will statically allocate all the DDBs for all the
planes on all the pipes.
However, when SOS or UOS wants to use Y/Yf tiled modifier, the watermark
required will exceed the DDBs allocated by GVT-g, thus causes some
display issues.

So in this patch, we removed the supports of the Y/Yf tiled modifiers
for both SOS and UOS when plane restriction is enabled. And a consequence
is that the RBC will be disabled since _CCS modifiers will no longer be
supported.

Signed-off-by: Min He <min.he@intel.com>
Reviewed-by: Zhao Yakui <yakui.zhao@intel.com>
Reviewed-by: Fei Jiang <fei.jiang@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index fb18154..ac8c6fd 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -13563,10 +13563,7 @@ intel_skl_plane_create(struct drm_i915_private *dev_priv, enum pipe pipe,
 	intel_plane->id = plane;
 	plane_formats = skl_primary_formats;
 
-	if (pipe < PIPE_C)
-		modifiers = skl_format_modifiers_ccs;
-	else
-		modifiers = skl_format_modifiers_noccs;
+	modifiers = i9xx_format_modifiers;
 
 	num_formats = ARRAY_SIZE(skl_primary_formats);
 
-- 
2.7.4

